polaris.shopify.com/pages/api/v0/tokens/[tokens].tsx

```diff
@@ -97,67 +60,46 @@ const formatTokenGroup = (tokenGroup: TokenGroup, format: Format) => {
       .join('\n');
   }
 
-  return tokenGroup;
+  return tokenValues;
 };
 
 const handler = async (req: NextApiRequest, res: NextApiResponse) => {
-  const formatParam = (req.query.format || 'json') as Format;
-  const schemeParam = (req.query.scheme || 'light') as Scheme;
+  const formatParam = isFormat(req.query.format) ? req.query.format : 'json';
+  const schemeParam = isScheme(req.query.scheme) ? req.query.scheme : 'light';
 
   if (typeof formatParam === 'string' && typeof schemeParam === 'string') {
-    const colorSchemeTokenGroup = colorSchemeMap[schemeParam];
-    const tokenGroupParam: any = req.query.tokens || '';
-    let tokenGroups: string[] = [];
+    const tokenGroupParam = (req.query.tokens || '') as TokenGroupKey;
+    let tokenData: TokenGroup = {};
 
     // Determine which list(s) we are querying for based on the token param
     if (tokenGroupParam === 'all') {
-      tokenGroups = tokenGroups.concat(staticTokenGroupKeys);
-    }
+      staticTokenGroupKeys.forEach((group) => {
+        const tokenGroup: TokenGroup = tokens[group] || {};
 
-    if (tokenGroupParam === 'all' || tokenGroupParam === 'colors') {
-      tokenGroups.push(colorSchemeTokenGroup);
+        tokenData = {...tokenData, ...tokenGroup};
+      });
     }
 
-    if (staticTokenGroupKeys.includes(tokenGroupParam)) {
-      tokenGroups.push(tokenGroupParam);
-    }
+    if (tokenGroupParam === 'all' || tokenGroupParam === 'colors') {
+      const colorSchemeTokenGroup: TokenGroup =
+        tokens.colorSchemes[schemeParam] || {};
 
-    if (tokenGroups.length === 0) {
-      res.status(400);
-      res.json({error: true, status: 400});
-      return;
+      tokenData = {...tokenData, ...colorSchemeTokenGroup};
     }
 
-    // Convert the list of urls to a group of requests
-    let requests: Promise<TokenGroup>[] = [];
-    try {
-      requests = tokenGroups.map((tokenFile) => {
-        const url = getGithubUrl(tokenFile, true);
-
-        if (url === null) {
-          throw new Error('Could not fetch tokens');
-        }
+    if (staticTokenGroupKeys.includes(tokenGroupParam as StaticTokenGroupKey)) {
+      const tokenGroupName = tokenGroupParam as StaticTokenGroupKey;
+      const tokenGroup: TokenGroup = tokens[tokenGroupName] || {};
```

Comment:
```suggestion
    if (isStaticTokenGroupKey(tokenGroupParam)) {
      const tokenGroup: TokenGroup = tokens[tokenGroupParam];
```

**suggestion (non-blocking):** If we keep the above `isStaticTokenGroupKey` type guard we can avoid the cast remove the fallback empty object.